src/theme/themeManager.ts

```diff
@@ -0,0 +1,58 @@
+export type CSSModuleClassNames = {
+	[key: string]: boolean;
+}
+
+export type AppliedClasses<T> = {
+	[P in keyof T]?: CSSModuleClassNames;
+}
+
+type Theme = {
+	[key: string]: string;
+}
+
+type CacheKey = {
+	baseThemeClasses: {};
+	overrideClasses?: {};
+};
+
+let currentTheme: Theme = {};
+let cacheMap = new Map<CacheKey, AppliedClasses<any>>();
+
+function addClassNameToMap(classMap: CSSModuleClassNames, classList: Theme, className: string) {
+	if (classList && classList.hasOwnProperty(className)) {
+		// split out the classname because css-module composition combines class names with a space
+		const generatedClassNames: string[] = classList[className].split(' ');
+		generatedClassNames.forEach((generatedClassName) => {
+			classMap[generatedClassName] = true;
+		});
+	}
+}
+
+export function setTheme(theme: {}) {
+	currentTheme = theme;
+	cacheMap = new Map<CacheKey, AppliedClasses<{}>>();
+};
+
+export function getClasses<T extends {}>(baseThemeClasses: T, overrideClasses?: {}): AppliedClasses<T> {
```

Comment:
Don't think so, because the return object is only keyed off of the baseThemeClasses. user might have put a superset in overrideClasses.